packages/edit-site/src/components/sidebar-edit-mode/template-panel/replace-template-button.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useState, useCallback } from '@wordpress/element';
+import { __experimentalBlockPatternsList as BlockPatternsList } from '@wordpress/block-editor';
+import { MenuItem, Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEntityRecord } from '@wordpress/core-data';
+import { useAsyncList } from '@wordpress/compose';
+import { serialize } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../../store';
+
+export default function ReplaceTemplateButton( {
+	onClick,
+	availableTemplates,
+} ) {
+	const [ showModal, setShowModal ] = useState( false );
+	const onClose = useCallback( () => {
+		setShowModal( false );
+	}, [] );
+
+	const { postId, postType } = useSelect( ( select ) => {
+		return {
+			postId: select( editSiteStore ).getEditedPostId(),
+			postType: select( editSiteStore ).getEditedPostType(),
+		};
+	}, [] );
+
+	const entity = useEntityRecord( 'postType', postType, postId );
+	const onTemplateSelect = async ( selectedTemplate ) => {
+		await entity.edit( {
+			blocks: selectedTemplate.blocks,
+			content: serialize( selectedTemplate.blocks ),
+		} );
+		onClose(); // Close the template suggestions modal first.
+		onClick();
```

Comment:
Suggestion: The `useEntityRecord` also subscribes to a store; that data isn't used here. It might be better to call `editEntityRecord` directly.

Question: Do we need an error handle here?

Reply:
Done. I don't see `editEntityRecord` being used with an error handler other times but that doesn't mean its not a good idea :)

Reply:
Like [this](https://github.com/WordPress/gutenberg/blob/5b286b720c14ae2f9c5832954a3c0f1d9a5fa13a/packages/edit-site/src/components/page-patterns/rename-menu-item.js#L38-L40)?

Reply:
@scruffian, you're correct. Edits don't make API request so it won't trigger an error. It's only `createEntityRecord` and `saveEditedEntityRecord`.

Sorry for the confusion.